File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/PutHotelResponse.java
package com.ll.hotel.domain.hotel.hotel.dto;

import com.ll.hotel.domain.hotel.hotel.entity.Hotel;
import com.ll.hotel.domain.review.review.dto.response.PresignedUrlsResponse;
import jakarta.validation.constraints.NotBlank;
import java.time.LocalDateTime;
import lombok.NonNull;

public record PutHotelResponse(
        long businessId,

        long hotelId,

        @NotBlank
        String hotelName,

        @NotBlank
        String hotelStatus,

        @NonNull
        LocalDateTime modifiedAt,

        PresignedUrlsResponse urlsResponse
) {
    public PutHotelResponse(Hotel hotel, PresignedUrlsResponse response) {
        this(
                hotel.getBusiness().getId(),
                hotel.getId(),
                hotel.getHotelName(),
                hotel.getHotelStatus().getValue(),
                hotel.getModifiedAt(),
                response
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/PutHotelResponse.kt
package com.ll.hotel.domain.hotel.hotel.dto

import com.ll.hotel.domain.hotel.hotel.entity.Hotel
import com.ll.hotel.domain.review.review.dto.response.PresignedUrlsResponse
import jakarta.validation.constraints.NotBlank
import java.time.LocalDateTime

data class PutHotelResponse(
    val businessId: Long,

    val hotelId: Long,

    @field:NotBlank
    val hotelName: String,

    @field:NotBlank
    val hotelStatus: String,

    val modifiedAt: LocalDateTime,

    val urlsResponse: PresignedUrlsResponse?
) {
    constructor(hotel: Hotel, response: PresignedUrlsResponse?) : this(
        hotel.business.id,
        hotel.id,
        hotel.hotelName,
        hotel.hotelStatus.value,
        hotel.modifiedAt,
        response
    )
}